CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/di/modules/RetrofitModule.kt

```diff
@@ -0,0 +1,38 @@
+package co.nimblehq.coroutine.di.modules
+
+import co.nimblehq.coroutine.BuildConfig
+import co.nimblehq.coroutine.data.service.ApiService
+import co.nimblehq.coroutine.data.service.providers.ApiServiceProvider
+import co.nimblehq.coroutine.data.service.providers.ConverterFactoryProvider
+import co.nimblehq.coroutine.data.service.providers.RetrofitProvider
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import okhttp3.OkHttpClient
+import retrofit2.Converter
+import retrofit2.Retrofit
+
+@Module
+@InstallIn(SingletonComponent::class)
+class RetrofitModule {
+
+    @Provides
+    fun provideBaseUrl() = BuildConfig.BASE_URL
+
+    @Provides
+    fun provideGsonConverterFactory(): Converter.Factory =
+        ConverterFactoryProvider.getGsonConverterFactory()
```

Comment:
I think we should use `moshiConverterFactory` instead 🤔 Gson was not maintained for a long time...